### PR TITLE
[4.0] Remove mod_multilangstatus tampering from mod_user

### PR DESCRIPTION
**Pull request by SharkyKZ**
Closes https://github.com/joomla/joomla-cms/issues/29948.

### Summary of Changes

Removes `mod_multilangstatus` tampering from `mod_user`.

### Testing Instructions

Create and publish a `Multilingual Status` module.

### Actual result BEFORE applying this Pull Request

Module created but keeps getting unpublished on every page load.

### Expected result AFTER applying this Pull Request

Module created and published.

### Documentation Changes Required

`Joomla\Module\Multilangstatus\Administrator\Helper\MultilangstatusAdminHelper` class is removed. If it's too late for that, it can be deprecated instead.

### CODE REVIEW

**Comment by infograf768** [comment]
As far as I understand, this breaks the automatic display of the module when multilang is ON.
I believe this was a real improvement in J4 for multilingual sites.

**Comment by SharkyKZ** [comment]
Yes, it stops unpublished modules being automatically published.

**Comment by infograf768** [comment]
I can't be in favor of that. As I said, this is definitely an improvement for all multilingual sites.

**Comment by SharkyKZ** [comment]
Hardcode it into the template if you want. Just stop messing with user data.

**Comment by infograf768** [comment]
We have enough explaining all the time in the forums how to use that module first when someone has a multilingual issue.

I was not the one who originally created that code and I don't know how to hard code it in atum.
This basically means that, if you want to get rid of it from mod_user, I am fine with it, but please provide also the solution to hardcode it in atum.

**Comment by brianteeman** [comment]
I agree with @infograf768 Dont just remove functionality without providing a replacement

**Comment by SharkyKZ** [comment]
Personally, I don't think there is a need for replacement for this. If you know how to enable Language Filter plugin, publishing the module shouldn't be that hard. But OK, some options:

1) Have the module published on all new sites.
2) Have the module published when installing multilingual sample data.
3) Have the module published when enabling Language Filter plugin.
4) Have the module embedded in the template.

Take your pick. Anything is better than what we have now.

bug2

**Comment by Bakual** [comment]
Why don't we have the module enabled by default, but display nothing when multilang is disabled? We have such behavior regulary in modules. Then it would automatically appear as soon as multilang is enabled. That would be the best approach imho.

A module should never automatically change its state. If it's unpublished, it has to stay unpublished, if it's published, it has to stay published. Also modules shouldn't interact with other modules, that's just plain wrong.
It should also not be hardcoded into the template, that would be equally wrong.

**Comment by Bakual** [comment]
Just for reference a PR which I did last year. Turns out the code was just moved in between and the PR mistakenly closed.
https://github.com/joomla/joomla-cms/pull/24341
It's more or less exactly the same as this one here.

**Comment by Bakual** [comment]
@SharkyKZ If you add the following lines to administrator/modules/mod_multilangstatus/mod_multilangstatus.php at line 13 (after the use statement), then you got option 1 but the module doesn't show up until the language filter plugin is enabled. Similar to other modules that don't show up if there is nothing to show.
````
use Joomla\CMS\Language\Multilanguage;

if (!Multilanguage::isEnabled())
{
	return;
}
````

**Comment by SharkyKZ** [comment]
PR updated. Using the check from `com_languages`.

File: administrator/modules/mod_multilangstatus/mod_multilangstatus.php [hunk 1]
```diff
@@ -10,5 +10,9 @@
 defined('_JEXEC') or die;
 
 use Joomla\CMS\Helper\ModuleHelper;
+use Joomla\CMS\Language\Multilanguage;
+use Joomla\Component\Languages\Administrator\Helper\MultilangstatusHelper;
+
+$mutilanguageEnabled = Multilanguage::isEnabled() && MultilangstatusHelper::getLangswitchers();
```
Comment (infograf768) [inline administrator/modules/mod_multilangstatus/mod_multilangstatus.php]: Please take off `MultilangstatusHelper::getLangswitchers()`. Checking for `Multilanguage::isEnabled()` is enough.

Comment (infograf768) [reply administrator/modules/mod_multilangstatus/mod_multilangstatus.php]: The multilangstatus module module takes care of any problem with the switcher.

Screen Shot 2020-07-17 at 07 25 04

Comment (infograf768) [reply administrator/modules/mod_multilangstatus/mod_multilangstatus.php]: Therefore no need to define `$mutilanguageEnabled`
Just modify default.php
```
use Joomla\CMS\HTML\HTMLHelper;
use Joomla\CMS\Language\Multilanguage;
use Joomla\CMS\Language\Text;
use Joomla\CMS\Router\Route;

if (!Multilanguage::isEnabled())
{
	return;
}
```

Comment (SharkyKZ) [reply administrator/modules/mod_multilangstatus/mod_multilangstatus.php]: If you insist.

**Comment by infograf768** [comment]
I don't understand anything to your changes. They look much more complex than our suggestion. Any reason to choose such a code?
if you insist on this, please correct typo `$mutilanguageEnabled` to `$multilanguageEnabled`

**Comment by SharkyKZ** [comment]
It's the same but with dependencies passed to `Multilanguage::isEnabled()`. Typo fixed.

**Comment by infograf768** [comment]
Do you mean that we should change everywhere `Multilanguage::isEnabled()` to `Multilanguage::isEnabled($app, Factory::getContainer()->get(DatabaseInterface::class));` ?

A simple grep gives me 134 occurrences of that code present in 88 files in J4

**Comment by SharkyKZ** [comment]
That seems to be the way going forward. But there's no need to rush replacing every existing instance now.

**Comment by richard67** [comment]
@wilsonge Any opinion on the class removal (or deprecation) mentioned in the "Documentation Changes Required" section of the description?

**Comment by infograf768** [comment]
IMHO, as it does not exist in J3, no need to deprecate it.
People who are using beta versions should only use them for tests.

**Comment by vijaykhollam** [comment]
I have tested this item :white_check_mark: successfully on f2ff1d4d81be05754e1aaf3afb9a1d64a4354d58<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/30113">issues.joomla.org/tracker/joomla-cms/30113</a>.</sub>

**Comment by Kaustubharas** [comment]
I have tested this item :white_check_mark: successfully on f2ff1d4d81be05754e1aaf3afb9a1d64a4354d58<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/30113">issues.joomla.org/tracker/joomla-cms/30113</a>.</sub>

**Comment by alikon** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/30113">issues.joomla.org/tracker/joomla-cms/30113</a>.</sub>

**Comment by HLeithner** [comment]
Thanks

**Pull request merged**